### PR TITLE
#9046 Fix unlinked extension console bug

### LINKED ISSUE #9046: Linking Extension after clearing PixieBrix token hangs
## Describe the bug

After clearing the PixieBrix token, the Extension Console hangs instead of showing the `Link Extension` button. Refreshing the Extension Console does show the button

## To Reproduce

1. Navigate to the Extension Console 
2. Click on `Settings` in the nav sidebar
3. Scroll down and click the clear PixieBrix button
4. Click `Mods` in the nav sidebar

## Actual behavior

The loading icon shows and never clears. Refreshing the page shows the link button

## Expected behavior

The link button should show

## Screenshots/Loom

https://www.loom.com/share/5629c1f527a54b16bf17ecab9274606a

## Desktop (please complete the following information):

- Extension Version: 
    - happens on main
    - does not happen on 2.1.0-beta.3
    - only difference is this commit: https://github.com/pixiebrix/pixiebrix-extension/commit/12ff6ef44b55d7d5fe14edd83f89d5bc4ec58d44 

## Notes

- https://app.rainforestqa.com/tests/379024?envId=61913&job=56965131&run=1915799&jobGroup=21586223&step=38

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Fixes #9046 

## Remaining Work
- [x] Add tests for `ModsPage` to make sure it's rendering the right top-level views (GettingStarted, Onboarding, Empty, Table content, etc.)

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

File: src/extensionConsole/pages/mods/ModsPage.test.tsx [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+      {
+        setupRedux(dispatch) {
+          dispatch(modsPageSlice.actions.setIsLoadingData(true));
+        },
+      },
+    );
+
+    await waitFor(() => {
+      expect(screen.getByTestId("loader")).toBeInTheDocument();
+    });
```
Comment (grahamlangford) [inline src/extensionConsole/pages/mods/ModsPage.test.tsx:93]: waitFor isn't doing anything (tested locally)
```suggestion
      expect(screen.getByTestId("loader")).toBeInTheDocument();
```

File: src/extensionConsole/pages/mods/ModsPage.test.tsx [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+
+    render(
+      <DeploymentsProvider>
+        <ModsPage />
+      </DeploymentsProvider>,
+    );
+
+    await waitFor(() => {
+      expect(screen.getByTestId("loader")).toBeInTheDocument();
+    });
```
Comment (grahamlangford) [inline src/extensionConsole/pages/mods/ModsPage.test.tsx:107]:
```suggestion
      expect(screen.getByTestId("loader")).toBeInTheDocument();
```

File: src/extensionConsole/pages/mods/ModsPage.test.tsx [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+      {
+        setupRedux(dispatch) {
+          dispatch(modsPageSlice.actions.setIsLoadingData(false));
+        },
+      },
+    );
+
+    await waitFor(() => {
+      expect(screen.getByText("An error occurred")).toBeInTheDocument();
+    });
```
Comment (grahamlangford) [inline src/extensionConsole/pages/mods/ModsPage.test.tsx:128]:
```suggestion
      expect(screen.getByText("An error occurred")).toBeInTheDocument();
```

File: src/extensionConsole/pages/mods/ModsPage.test.tsx [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
```
Comment (grahamlangford) [inline src/extensionConsole/pages/mods/ModsPage.test.tsx:148]: NIT: This is the "correct" way to write this:
```suggestion
    await expect(screen.findByText("An error occurred")).resolves.toBeInTheDocument();
```

File: src/extensionConsole/pages/mods/ModsPage.test.tsx [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+          );
+        },
+      },
+    );
+
+    await waitFor(() => {
+      expect(
+        screen.getByText("Welcome to the PixieBrix Extension Console"),
+      ).toBeInTheDocument();
+    });
```
Comment (grahamlangford) [inline src/extensionConsole/pages/mods/ModsPage.test.tsx:171]:
```suggestion
      expect(
        screen.getByText("Welcome to the PixieBrix Extension Console"),
      ).toBeInTheDocument();
```

File: src/extensionConsole/pages/mods/ModsPage.test.tsx [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+              isReactivate: false,
+            }),
+          );
+          dispatch(modsPageSlice.actions.setIsLoadingData(false));
+        },
+      },
+    );
+
+    waitForAct();
+    expect(screen.getByText("Test Mod 1")).toBeInTheDocument();
```
Comment (grahamlangford) [inline src/extensionConsole/pages/mods/ModsPage.test.tsx:219]: `waitForAct` is unnecessary
```suggestion
    expect(screen.getByText("Test Mod 1")).toBeInTheDocument();
```

File: src/extensionConsole/pages/mods/ModsPage.test.tsx [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+              screen: "extensionConsole",
+              isReactivate: false,
+            }),
+          );
+        },
+      },
+    );
+
+    waitForAct();
+
```
Comment (grahamlangford) [inline src/extensionConsole/pages/mods/ModsPage.test.tsx:260]:
```suggestion
```

File: src/extensionConsole/pages/mods/ModsPage.test.tsx [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+        <ModsPage />
+      </DeploymentsProvider>,
+      {
+        setupRedux(dispatch) {
+          dispatch(modsPageSlice.actions.setIsLoadingData(false));
+        },
+      },
+    );
+
+    waitForAct();
```
Comment (grahamlangford) [inline src/extensionConsole/pages/mods/ModsPage.test.tsx:282]:
```suggestion
```

File: src/testUtils/testHelpers.tsx [hunk 9]
```diff
@@ -52,15 +52,34 @@ import {
...
+ * these cases, waitForEffect causes the render to hang indefinitely.
+ *
+ *  @see waitForEffect
+ */
+export const waitForAct = () =>
+  // eslint-disable-next-line testing-library/no-unnecessary-act,@typescript-eslint/no-confusing-void-expression -- hack for testing some asynchronous code that the standard utilities have proven inadequate
+  act(() => {
+    // Awaiting the state update
+  });
+
```
Comment (grahamlangford) [inline src/testUtils/testHelpers.tsx:82]: I'm not sure what weird behavior you were seeing locally, but this doesn't do anything

File: src/extensionConsole/store.ts [hunk 10]
```diff
@@ -81,7 +81,7 @@ export type RootState = AuthRootState &
   SessionChangesRootState;
 
 const conditionalMiddleware: Middleware[] = [];
-if (typeof createLogger === "function") {
+if (typeof createLogger === "function" && process.env.JEST_WORKER_ID == null) {
```
Comment (BLoe) [inline src/extensionConsole/store.ts:84]: @grahamlangford Does this make sense? I was getting annoyed with all the log spam in my test just trying to scroll down to the results/error message. I don't think we need the redux logger for anything in jest tests.

**Review by grahamlangford — APPROVED** [review]

File: src/extensionConsole/store.ts (hunk 10 above)
Comment (grahamlangford) [reply src/extensionConsole/store.ts:84]: I've never had an issue with that
